code/solutions/resource-dependencies/resource-dependencies-challenge.yaml

```diff
@@ -0,0 +1,42 @@
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: AWS CloudFormation workshop - Resource Dependencies Challenge Lab (uksb-1q9p31idr).
+
+Parameters:
+  S3BucketName:
+    Description: Enter a unique name for S3 bucket.
```

Comment:
Can you also consider not specifying an S3 bucket name - and not specifying an SNS topic name in the workshop content - so that those are created automatically in these examples?  Otherwise, I recommend to consider adding input validation in templates for e.g., S3, SNS resources, to facilitate user input and fail-fast.

If you choose to have S3 and SNS resource names determined automatically, you will also need to reflect this in the lab content, where you ask the user to specify unique values, and remove these occurrences.

Reply:
Also, if you choose to have resource names being created automatically, consider replacing relevant properties that you remove with `Tags` sections (and have an e.g., `Name` value set to `resource-dependencies-workshop` or similar, so that the `Properties` section is still shown in the hierarchy.

Reply:
Changed the content and code as per the suggestion, removed the parameter section and added `Tags` property